frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-menu-action.ts

```diff
@@ -0,0 +1,35 @@
+import { TFunction } from 'i18next';
+
+import { asAccessReview } from '@console/internal/components/utils';
+import { K8sKind } from '@console/internal/module/k8s';
+
+import { StoragePoolKind } from '../../types';
+import { updateBlockPoolModal } from '../modals/block-pool-modal/update-block-pool-modal';
+import { deleteBlockPoolModal } from '../modals/block-pool-modal/delete-block-pool-modal';
+
+const editBlockPool = (kindObj: K8sKind, blockPoolConfig: StoragePoolKind, _, customData) => {
+  const t: TFunction = customData?.tFunction;
+  return {
+    labelKey: t('ceph-storage-plugin~Edit BlockPool'),
+    callback: () => updateBlockPoolModal({ kindObj, blockPoolConfig }),
+    accessReview: asAccessReview(kindObj, blockPoolConfig, 'patch'),
+  };
+};
+
+const deleteBlockPool = (kindObj: K8sKind, blockPoolConfig: StoragePoolKind, _, customData) => {
+  const t: TFunction = customData?.tFunction;
+  return {
+    labelKey: t('ceph-storage-plugin~Delete BlockPool'),
+    callback: () => deleteBlockPoolModal({ kindObj, blockPoolConfig }),
+    accessReview: asAccessReview(kindObj, blockPoolConfig, 'patch'),
+  };
+};
+
+export const menuActions = [editBlockPool, deleteBlockPool];
+
+export const menuActionCreator = (
+  kindObj: K8sKind,
+  blockPoolConfig: StoragePoolKind,
+  resource?: any,
+  customData?: any,
+) => menuActions.map((action) => action(kindObj, blockPoolConfig, resource, customData));
```

Comment:
Not optimizing deleteBlockPool and editBlockPool function parameters, it is called from two different places. For the overview page, it is invoked using the menuActionCreator function by 
```
<DetailsPage
      {...props}
      pagesFor={pagesFor}
      menuActions={menuActionCreator}
      customData={{ tFunction: t }}
    />
```

And for list page, it is invoked by 

```
 <ResourceKebab
          actions={menuActions}
          kind={referenceFor(obj)}
          resource={obj}
   >
```
I can't shrink the parameters here